### PR TITLE
Explicitly mark the AsyncSequence iterators as non Sendable

### PR DESCRIPTION
**Pull request by fabianfett**

### CODE REVIEW

File: Sources/PostgresNIO/New/PostgresNotificationSequence.swift [hunk 1]
```diff
@@ -20,3 +20,6 @@ public struct PostgresNotificationSequence: AsyncSequence, Sendable {
         }
     }
 }
+
+@available(*, unavailable)
+extension PostgresNotificationSequence.AsyncIterator :Sendable {}
```
Comment (fabianfett) [inline Sources/PostgresNIO/New/PostgresNotificationSequence.swift:25]:
```suggestion
extension PostgresNotificationSequence.AsyncIterator: Sendable {}
```

File: Sources/PostgresNIO/New/PostgresRowSequence.swift [hunk 2]
```diff
@@ -56,6 +56,9 @@ extension PostgresRowSequence {
     }
 }
 
+@available(*, unavailable)
+extension PostgresRowSequence.AsyncIterator :Sendable {}
```
Comment (fabianfett) [inline Sources/PostgresNIO/New/PostgresRowSequence.swift:60]:
```suggestion
extension PostgresRowSequence.AsyncIterator: Sendable {}
```

File: Package.swift [hunk 3]
```diff
@@ -2,7 +2,8 @@
 import PackageDescription
 
 let swiftSettings: [SwiftSetting] = [
-    .enableUpcomingFeature("StrictConcurrency")
+    .enableUpcomingFeature("StrictConcurrency"),
+    .unsafeFlags(["-require-explicit-sendable"]),
```
Comment (gwynne) [inline Package.swift:6]: Can't leave an unsafe flag in a versioned library package

**Review by gwynne — CHANGES_REQUESTED** [review]

File: Package.swift (hunk 3 above)
Comment (fabianfett) [reply Package.swift:6]:
```suggestion
```

**Review by gwynne — APPROVED** [review]

**Pull request merged**